spa-react/src/components/Chat/Chat.tsx

```diff
@@ -0,0 +1,163 @@
+import { collection, deleteDoc, doc, onSnapshot } from "firebase/firestore";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { useAppDispatch, useAppSelector } from "src/store/hooks";
+import {
+  getChatId,
+  getMessages,
+  getReceiverEmail,
+  getUserEmail,
+} from "src/store/selectors";
+import {
+  resetAllMessages,
+  setNewMessage,
+  setValue,
+} from "src/store/slices/chatSlice";
+import { db } from "src/utils/firebase";
+import { ReactComponent as MessageImg } from "../../assets/message.svg";
+import { InputChat } from "../InputChat/InputChat";
+import { Message } from "../Message/Message";
+import { MessageWithoutAvatar } from "../Message/MessageWithoutAvatar";
+import { IMessage } from "../Message/types";
+import { ChatsSectionSC } from "../Sidebar/style";
+import {
+  ChatSectionSC,
+  ContainerMessagesSC,
+  ImageContainerSC,
+  TextSC,
+} from "./style";
+
+export const Chat = () => {
+  const { reset } = useForm();
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const chatId = useAppSelector(getChatId);
+  const messages = useAppSelector(getMessages);
+  const userEmail = useAppSelector(getUserEmail);
+  const recieverEmail = useAppSelector(getReceiverEmail);
+
+  const [sent, setSent] = useState<IMessage[]>();
+  const [receive, setReceive] = useState<IMessage[]>();
+
+  const [edit, setEdit] = useState<boolean>(false);
+  const [messageId, setMessageId] = useState<string>("");
+
+  useEffect(() => {
+    if (userEmail) {
+      onSnapshot(collection(db, "messages"), (querySnapshot) => {
+        let messages: any[] = [];
+
+        let filteredSent: IMessage[] = [];
+        let filteredReceive: IMessage[] = [];
+
+        querySnapshot.forEach((doc) => {
+          messages.push(doc.data());
+        });
+
+        messages?.map((message) => {
+          return message.receivers.map((receiver: string) => {
+            return (
+              (receiver === recieverEmail && filteredSent.push(message)) ||
+              (receiver === userEmail && filteredReceive.push(message))
+            );
+          });
+        });
+
+        setSent(filteredSent);
+        setReceive(filteredReceive);
+      });
+    }
+  }, [recieverEmail, userEmail, dispatch]);
+
+  useEffect(() => {
+    dispatch(resetAllMessages());
+
+    sent?.map((item) => {
```

Comment:
looks strange. Why do you need here useEffect for sending of messages?!

Reply:
we put messages which we sent to firestore , and then take them from firestore and put to the redux. So this useEffect take messages which we sent and which were sent to us